src/perform_step/low_storage_rk_perform_step.jl

```diff
@@ -182,3 +182,90 @@ end
 
   f(k, u, p, t+dt)
 end
+
+
+
+# 2R+ low storage methods
+function initialize!(integrator,cache::LowStorageRK2RPConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 1
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPConstantCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack A₁,Bₗ,B̂ₗ,Bᵢ,B̂ᵢ,Cᵢ = cache
+
+  Xᵢ  = uprev
+  k   = fsalfirst
+  tmp = zero(uprev)
+
+  #stages 1 to s-1
+  for i in eachindex(A₁)
+    tmp = tmp + (Bᵢ[i] - B̂ᵢ[i])*dt*k
+    u   = Xᵢ + A₁[i]*dt*k
+    Xᵢ  = u + (Bᵢ[i] - A₁[i])*dt*k
+    k   = f(u, p, t + Cᵢ[i]*dt)
+  end
+
+  #last stage
+  tmp = tmp + (Bₗ - B̂ₗ)*dt*k
+  u   = Xᵢ  + Bₗ*dt*k
+
+  #Error estimate
+  if integrator.opts.adaptive
+    atmp = calculate_residuals(tmp, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm, t)
+    integrator.EEst = integrator.opts.internalnorm(atmp,t)
+  end
+
+  integrator.k[1] = integrator.fsalfirst
+  integrator.fsallast = f(u, p, t+dt) # For interpolation, then FSAL'd
+  integrator.u = u
+end
+
+function initialize!(integrator,cache::LowStorageRK2RPCache)
+  integrator.kshortsize = 2
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.fsalfirst = cache.fsalfirst
+  integrator.fsallast = cache.k
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst, integrator.uprev, integrator.p, integrator.t)
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack k,tmp,atmp,Xᵢ = cache
+  @unpack A₁,Bₗ,B̂ₗ,Bᵢ,B̂ᵢ,Cᵢ = cache.tab
+
+  @. Xᵢ = uprev
+  @. k  = fsalfirst
+  @. tmp = zero(u)
+
+  #stages 1 to s-1
+  for i in eachindex(A₁)
+    @. tmp = tmp + (Bᵢ[i] - B̂ᵢ[i])*dt*k
+    @. u   = Xᵢ + A₁[i]*dt*k
+    @. Xᵢ  = u + (Bᵢ[i] - A₁[i])*dt*k
+    f(k, u, p, t + Cᵢ[i]*dt)
+  end
+
+  #last stage
+  @. tmp = tmp + (Bₗ - B̂ₗ)*dt*k
+  @. u   = Xᵢ  + Bₗ*dt*k
+
+  #Error estimate
```

Comment:
If we want to reduce the memory requirements (and number of operations), the computation of the error estimator above (what is now called `tmp`) should be avoided if `adaptive==false`. In that case, the additional array could also be aliased.

Reply:
This hasn't been addressed yet?

Reply:
It's done as directed. I don't know why it's not showing outdated.